adot-testbed/app/src/test/java/software/amazon/adot/testbed/LogsTests.java

```diff
@@ -65,61 +73,164 @@ private GenericContainer<?> createAndStartCollector(String configFilePath, Strin
-            .withEnv(envVariables);
+            .withEnv(envVariables)
+            .withClasspathResourceMapping("/logs", "/logs", BindMode.READ_WRITE)
+            .withCommand("--config", "/etc/collector/config.yaml", "--feature-gates=+adot.filelog.receiver,+adot.awscloudwatchlogs.exporter,+adot.file_storage.extension");
 
-        //Mount the log file for the file log receiver to parse
-        collector.withCopyFileToContainer(MountableFile.forClasspathResource(logFilePath), logFilePath );
+       //Mount the Temp directory
+        collector.withFileSystemBind(logDirectory.toString(),"/tempLogs", BindMode.READ_WRITE);
 
         collector.start();
-        collector.waitingFor(Wait.forHealthcheck());
         return collector;
     }
 
     @Test
     void testSyslog() throws Exception {
         String logStreamName = "rfcsyslog-logstream-" + uniqueID;
-        collector = createAndStartCollector("/configurations/config-rfcsyslog.yaml", "/logs/RFC5424.log", logStreamName);
+        collector = createAndStartCollector("/configurations/config-rfcsyslog.yaml", logStreamName);
 
-        validateLogs(logStreamName , "/logs/RFC5424.log");
+        List<String> logFilePaths = new ArrayList<>();
+        logFilePaths.add("/logs/RFC5424.log");
+        validateLogs(logStreamName , logFilePaths);
         collector.stop();
     }
 
     @Test
     void testLog4j() throws Exception {
         String logStreamName = "log4j-logstream-" + uniqueID;
-        collector = createAndStartCollector("/configurations/config-log4j.yaml", "/logs/log4j.log", logStreamName);
+        collector = createAndStartCollector("/configurations/config-log4j.yaml", logStreamName);
 
-        validateLogs(logStreamName , "/logs/log4j.log");
+        List<String> logFilePaths = new ArrayList<>();
+        logFilePaths.add("/logs/log4j.log");
+        validateLogs(logStreamName , logFilePaths);
         collector.stop();
     }
 
     @Test
     void testJson() throws Exception {
         String logStreamName = "json-logstream-" + uniqueID;
-        collector = createAndStartCollector("/configurations/config-json.yaml", "/logs/testingJSON.log", logStreamName);
+        collector = createAndStartCollector("/configurations/config-json.yaml", logStreamName);
+
+        List<String> logFilePaths = new ArrayList<>();
+        logFilePaths.add("/logs/testingJSON.log");
+        validateLogs(logStreamName , logFilePaths);
+        collector.stop();
+    }
+
+    @Test
+    void testCollectorRestartStorageExtension() throws Exception {
+        String logStreamName = "storageExtension-logstream-" + uniqueID;
+        collector = createAndStartCollector("/configurations/config-storageExtension.yaml", logStreamName);
+        File tempFile = new File(logDirectory.toString(), "storageExtension.log");
+        Thread.sleep(5000);
```

Comment:
Why sleep here?

Reply:
The sleep is similar as [here](https://github.com/aws-observability/aws-otel-test-framework/pull/1427#discussion_r1356420926), its just to wait for the collector to start watching the file.